app/main/views/report_requests.py

```diff
@@ -1,10 +1,88 @@
-from flask import abort, send_file
+from flask import abort, jsonify, render_template, send_file, url_for
 from flask_login import current_user
+from notifications_python_client.errors import HTTPError
+from werkzeug.utils import redirect
 
-from app.constants import REPORT_REQUEST_STORED
+from app import current_service
+from app.constants import REPORT_REQUEST_FAILED, REPORT_REQUEST_STORED
 from app.main import main
 from app.models.report_request import ReportRequest
-from app.utils.user import user_is_platform_admin
+from app.utils.user import user_has_permissions, user_is_platform_admin
+
+
+@main.route("/services/<uuid:service_id>/download-report/<uuid:report_request_id>", methods=["GET"])
+@user_has_permissions()
+def report_request(service_id, report_request_id):
+    try:
+        report_request = ReportRequest.from_id(service_id, report_request_id)
+    except HTTPError as e:
+        # if there is no request ID, endpoint will return an error
+        # so we want to show a useful page instead of a 404
+        if e.status_code == 404:
+            return render_template(
+                "views/csv-report/unavailable.html",
+            )
+    else:
+        if report_request.user_id != current_user.id:
+            abort(403)
+        # if they refresh that page manually before JS does
+        # or they copy and paste that url in manually or bookmark it
+        # we redirect them to the download page
+        if report_request.status == REPORT_REQUEST_STORED:
+            return redirect(
+                url_for(
+                    "main.report_ready",
+                    service_id=service_id,
+                    report_request_id=report_request.id,
+                )
+            )
+
+        if report_request.status == REPORT_REQUEST_FAILED:
+            return render_template(
+                "views/csv-report/error.html",
+            )
+
+    return render_template(
+        "views/csv-report/preparing.html",
+        retention_period=current_service.get_days_of_retention("email"),
```

Comment:
```suggestion
        retention_period=current_service.get_days_of_retention(report_type),
```
let's keep like the original implementation [here](https://github.com/alphagov/notifications-admin/blob/dcf716a72b14b92074bd495b1e3d34fb2d1fb518/app/main/views/dashboard.py#L265)

Reply:
can make it `message_type`, not `report_type` as report_type is 'notifications_report`.
i think get_days_of_retention logic does fallback as well